packages/yourloops/pages/training/training.tsx

```diff
@@ -0,0 +1,159 @@
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { useMemo } from 'react'
+
+import { useAuth } from '../../lib/auth'
+import { Trans, useTranslation } from 'react-i18next'
+import _ from 'lodash'
+import { Profile } from '../../models/user'
+import bows from 'bows'
+import { useHistory } from 'react-router-dom'
+import { HistoryState } from '../../models/generic'
+import Button from '@material-ui/core/Button'
+import Grid from '@material-ui/core/Grid'
+import Card from '@material-ui/core/Card'
+import CardMedia from '@material-ui/core/CardMedia'
+import Box from '@material-ui/core/Box'
+import appConfig from '../../lib/config'
+import CardContent from '@material-ui/core/CardContent'
+import Container from '@material-ui/core/Container'
+import { makeStyles, Theme } from '@material-ui/core/styles'
+import Link from '@material-ui/core/Link'
+import diabeloopUrls from '../../lib/diabeloop-url'
+
+const style = makeStyles((theme: Theme) => {
+  return {
+    mainContainer: {
+      [theme.breakpoints.down('xs')]: {
+        padding: 0
+      }
+    },
+    card: {
+      textAlign: 'center',
+      padding: theme.spacing(4)
+    },
+    cardContent: {
+      marginLeft: theme.spacing(2),
+      marginRight: theme.spacing(2),
+      [theme.breakpoints.down('sm')]: {
+        marginLeft: theme.spacing(0),
+        marginRight: theme.spacing(0)
+      }
+    },
+    centeredGrid: {
+      minHeight: '90vh'
+    },
+    buttons: {
+      display: 'flex',
+      justifyContent: 'space-around',
+      marginTop: theme.spacing(2),
+      [theme.breakpoints.down('sm')]: {
+        justifyContent: 'space-between'
+      }
+    }
+  }
+}, { name: 'ylp-component-consent' })
+/**
+ * Training Page
+ */
+function TrainingPage(): JSX.Element {
+  const { t, i18n } = useTranslation('yourloops')
+  const auth = useAuth()
+  const log = useMemo(() => bows('consent'), [])
+  const historyHook = useHistory<HistoryState>()
+  const fromPath = useMemo(() => historyHook.location.state?.from?.pathname, [historyHook])
+  const user = auth.user
+  const classes = style()
+
+  const ackTraining = (): void => {
+    const now = new Date().toISOString()
+    const updatedProfile = _.cloneDeep(user.profile ?? {}) as Profile
```

Comment:
We tend to avoid using loadash. Is it really necessary to clone deep the profile? 
Are there risks in using: `const updatedProfile = user.profile ?? {} as Profile`?

Reply:
I'm afraid that modifying the profile will trigger rendering in other parts of the product ... not what i want